operator/pkg/ciliumendpointbatch/ceptocebmap.go

```diff
@@ -69,3 +69,72 @@ func (c *cepToCebMapping) count() int {
 	defer c.cepMutex.RUnlock()
 	return len(c.cacheCep)
 }
+
+// desiredCebMapping is used to map CiliumEndpointBatch name to cebTracker object.
+// This map is used by the CEB manager, primarily used for storing and retrieving
+// the desired CEBs by thread-safe.
+// This map is protected by lock for consistent and concurrent access.
+type desiredCebMapping struct {
+	cebMutex lock.RWMutex
+	// desiredCebs is used to map cebName to cebTracker object.
+	desiredCebs map[CEBName]*cebTracker
+}
+
+// Creates and intializes the new desiredCebMapping
+func newDesiredCebMap() *desiredCebMapping {
+	return &desiredCebMapping{
+		desiredCebs: make(map[CEBName]*cebTracker),
+	}
+}
+
+// Insert the CEB tracker in map
+func (c *desiredCebMapping) insert(cebName string, ceb *cebTracker) {
+	c.cebMutex.Lock()
+	defer c.cebMutex.Unlock()
+	c.desiredCebs[CEBName(cebName)] = ceb
+}
+
+// Remove the CEB tracker from map
+func (c *desiredCebMapping) deleteCeb(cebName string) {
+	c.cebMutex.Lock()
+	defer c.cebMutex.Unlock()
+	delete(c.desiredCebs, CEBName(cebName))
+}
+
+func (c *desiredCebMapping) get(cebName string) (*cebTracker, bool) {
+	c.cebMutex.RLock()
+	defer c.cebMutex.RUnlock()
+	ceb, ok := c.desiredCebs[CEBName(cebName)]
+	return ceb, ok
+}
+
+func (c *desiredCebMapping) getCeb(cebName string) *cebTracker {
```

Comment:
Confused why we need another function here, this seems exactly like the `get` function except that you ignore the ok return, two questions:
- why ignore?
- If you do need to ignore, can you do it in the caller?

Reply:
If the caller of `getCeb` function need only `cebTracker` object, `getCeb` can be used.  If we know for sure `cebTracker` is present in the `desiredCebMapping`, then affirmatively use  `getCeb`.

Reply:
can't you achieve the same thing with```ceb, _ := get()```?

Reply:
The whole point of having this to have syntax simpler
`ceb := getCeb()` or `len( getCeb().ceb.Endpoints)`

Reply:
I don't see `getCeb` used anywhere. I would prefer that we limit addition of not-used code until it's actually needed.

Reply:
Used here 
https://github.com/cilium/cilium/pull/17345/files#diff-d6fc34245039673287d7addb7ea1931c5b687da6e0141ead9725269e55fb1288R141-R143
and
https://github.com/cilium/cilium/pull/17345/files#diff-d6fc34245039673287d7addb7ea1931c5b687da6e0141ead9725269e55fb1288R410-R413